extensions/mssql/src/sparkFeature/dialog/dialogCommands.ts

```diff
@@ -49,31 +55,61 @@ export class OpenSparkJobSubmissionDialogCommand extends Command {
-		let displayList: string[] = new Array();
 		let connectionMap: Map<string, azdata.connection.Connection> = new Map();
+		let selectedHost: string = undefined;
+		let showConnectionDialog = false;
+
+		// Filter invalid connections
+		if (connectionList && connectionList.length > 0) {
+			connectionList = connectionList.filter(conn => conn.providerName === constants.sqlProviderName && (<MssqlOptions><any>conn.options).server);
+		}
+		// Prompt choice if we have active connections
 		if (connectionList && connectionList.length > 0) {
-			connectionList.forEach(conn => {
-				if (conn.providerName === constants.sqlProviderName) {
-					displayList.push(conn.options.host);
-					connectionMap.set(conn.options.host, conn);
-				}
+			let selectConnectionMsg = localize('selectOtherServer', 'Select other SQL Server');
+			let displayList: string[] = [];
+			if (connectionList && connectionList.length > 0) {
+				connectionList.forEach(conn => {
+					let options: MssqlOptions = <any>conn.options;
+					displayList.push(options.server);
+					connectionMap.set(options.server, conn);
+				});
+			}
+			displayList.push(selectConnectionMsg);
+
+			selectedHost = await vscode.window.showQuickPick(displayList, {
+				placeHolder:
+					localize('sparkJobSubmission_PleaseSelectSqlWithCluster',
+						"Please select SQL Server with big data cluster.")
 			});
+			if (selectedHost === selectConnectionMsg) {
+				showConnectionDialog = true;
+				selectedHost = undefined;
+			}
+		} else {
+			showConnectionDialog = true;
+		}
+
+		// Show connection dialog if still don't have a server
+		if (showConnectionDialog) {
+			let connection = await azdata.connection.openConnectionDialog([constants.sqlProviderName]);
+			if (connection) {
+				let options: MssqlOptions = <any>connection.options;
+				connectionMap.set(options.server, connection);
+				selectedHost = options.server;
+				// Wait an appropriate timeout so that the serverInfo object can populate...
+				await timeout(150);
+			}
 		}
 
-		let selectedHost: string = await vscode.window.showQuickPick(displayList, {
-			placeHolder:
-				localize('sparkJobSubmission_PleaseSelectSqlWithCluster',
-					"Please select SQL Server with big data cluster.")
-		});
-		let errorMsg = localize('sparkJobSubmission_NoSqlSelected', 'No Sql Server is selected.');
+		let errorMsg = localize('sparkJobSubmission_NoSqlSelected', 'No SQL Server is selected.');
 		if (!selectedHost) { throw new Error(errorMsg); }
 
 		let sqlConnection = connectionMap.get(selectedHost);
 		if (!sqlConnection) { throw new Error(errorMsg); }
 
 		let sqlClusterConnection = await SqlClusterLookUp.getSqlClusterConnection(sqlConnection);
 		if (!sqlClusterConnection) {
-			throw new Error(LocalizedConstants.sparkJobSubmissionNoSqlBigDataClusterFound);
+			throw new Error(localize('errorNotSqlBigDataCluster', 'The selected server is not a SQL Server big data cluster'));
```

Comment:
maybe: The selected server does not belong to a SQL Server big data cluster ?

Reply:
nice, will update